src/features/forms/components/transaction-form-item.tsx

```diff
@@ -133,67 +134,60 @@ function TransactionBuilder<TSchema extends Record<string, unknown>>({
-  disabled,
-  placeholder,
   transactionType,
-  launchModal,
-  ...props
 }: TransactionFormItemProps<TSchema>) {
-  const { register, setValue, getValues, trigger } = useFormContext<TSchema>()
-
-  const setTransaction = useCallback(
-    async (value: PathValue<TSchema, Path<TSchema>>) => {
-      setValue(field, value)
-      trigger(field)
-    },
-    [field, setValue, trigger]
-  )
-
-  const closeModal = useCallback(() => {
-    launchModal(undefined)
-  }, [launchModal])
+  const { setValue, watch, trigger } = useFormContext<TSchema>()
 
-  const fieldValue = getValues(field)
+  const fieldValue = watch(field)
   // TODO: NC - Make this better
-  const savedValues = fieldValue ? JSON.parse(fieldValue.toString()) : undefined
-  // TODO: NC - stringify is yuck
+  const savedValues = useMemo(() => (fieldValue ? JSON.parse(fieldValue.toString()) : undefined), [fieldValue])
+
+  // // TODO: NC - stringify is yuck
   const thign = savedValues ? Object.entries(savedValues).map(([key, value]) => ({ dt: key, dd: JSON.stringify(value) })) : []
 
-  const buildTransaction = useCallback(
-    (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
-      e.preventDefault()
-      launchModal(
-        <TransactionBuilder
-          transactionType={transactionType}
-          savedValues={savedValues}
-          onComplete={closeModal}
-          onAddTransaction={setTransaction}
-        />
-      )
-    },
-    [closeModal, launchModal, savedValues, setTransaction, transactionType]
-  )
+  const { open: openTransactionBuilderDialog, dialog: transactionBuilderDialog } = useDialogForm({
+    dialogHeader: 'Build Transaction',
+    dialogBody: (
+      props: DialogBodyProps<
+        { transactionType: algosdk.ABITransactionType; savedValues: Record<string, unknown> },
+        PathValue<TSchema, Path<TSchema>>
+      >
+    ) => (
+      <TransactionBuilder
+        data={{ transactionType: props.data.transactionType, savedValues: props.data.savedValues }}
+        onComplete={props.onCancel}
+        onAddTransaction={props.onSubmit}
+      />
+    ),
+  })
+
+  const openDialog = useCallback(async () => {
+    const transaction = await openTransactionBuilderDialog({
+      transactionType,
+      savedValues,
+    })
+    if (transaction) {
+      setValue(field, transaction)
+      await trigger(field)
+    }
+  }, [openTransactionBuilderDialog, transactionType, savedValues, setValue, field, trigger])
 
   return (
     <>
-      {!savedValues && <Button onClick={buildTransaction}>Create</Button>}
+      {!savedValues && (
+        <Button type="button" onClick={openDialog}>
+          Create
+        </Button>
+      )}
       {savedValues && (
         <>
           <DescriptionList items={thign} />
-          <Button onClick={buildTransaction}>Edit</Button>
+          <Button type="button" onClick={openDialog}>
+            Edit
+          </Button>
         </>
       )}
-      <FormItem {...props} field={field} disabled={disabled}>
```

Comment:
@neilcampbell I removed this hidden input because the parent form failed to reset with it being here. Do we need this input?

Reply:
Yep, the transaction information is JSON serialised into that input and that's the transaction is materialised.
This is unless you've changed the logic to do it differently?